src/Microsoft.TestPlatform.CoreUtilities/Utilities/MulticastDelegateUtilities.cs

```diff
@@ -22,40 +22,93 @@ public static class MulticastDelegateUtilities
         /// <param name="sender">Sender to use when raising the event.</param>
         /// <param name="args">Arguments to provide.</param>
         /// <param name="traceDisplayName">Name to use when tracing out errors.</param>
+        // Using [CallerMemberName] for the traceDisplayName is a possibility, but in few places we call through other
+        // methods until we reach here. And it would change the public API.
         public static void SafeInvoke(this Delegate delegates, object sender, EventArgs args, string traceDisplayName)
+        {
+            SafeInvoke(delegates, sender, (object)args, traceDisplayName);
+        }
+
+        /// <summary>
+        /// Invokes each of the subscribers of the event and handles exceptions which are thrown
+        /// ensuring that each handler is invoked even if one throws.
+        /// </summary>
+        /// <param name="delegates">Event handler to invoke.</param>
+        /// <param name="sender">Sender to use when raising the event.</param>
+        /// <param name="args">Arguments to provide.</param>
+        /// <param name="traceDisplayName">Name to use when tracing out errors.</param>
+        // Using [CallerMemberName] for the traceDisplayName is a possibility, but in few places we call through other
+        // methods until we reach here. And it would change the public API.
```

Comment:
For this one, you are adding the new method so I don't think that the comment `And it would change the public API.` makes total sense.

Reply:
The new method would have the same signature, and would change the meaning of the traceDisplayName slightly, because it would now capture the caller, instead of the provided value. You could still provided value, so on compile time the signature would stay compatible, but the assembly would not, because in order to be useful, parameter with [CallerMemberName] attribute needs to have default value (`= null`). This changes the public api, because it is not a backward compatible code change.